### PR TITLE
Add retry on update status

### PR DESCRIPTION
**Pull request by pigletfly**
Signed-off-by: pigletfly <wangbing.adam@gmail.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by RainbowMango** [comment]
/assign @mrlihanbo 
LGTM by the way.

File: pkg/controllers/binding/binding_controller.go [hunk 1]
```diff
@@ -275,8 +276,11 @@ func (c *ResourceBindingController) updateFullyAppliedCondition(binding *workv1a
...
 
-	meta.SetStatusCondition(&binding.Status.Conditions, newBindingFullyAppliedCondition)
-	err := c.Client.Status().Update(context.TODO(), binding)
-
-	return err
+	return retry.RetryOnConflict(retry.DefaultBackoff, func() (err error) {
+		if err = c.Get(context.TODO(), client.ObjectKey{Namespace: binding.Namespace, Name: binding.Name}, binding); err != nil {
+			return err
+		}
+		meta.SetStatusCondition(&binding.Status.Conditions, newBindingFullyAppliedCondition)
```
Comment (gy95) [inline pkg/controllers/binding/binding_controller.go:283]: does it need to call function `SetStatusCondition` every time Retry? Is this specified? 😄

Comment (pigletfly) [reply pkg/controllers/binding/binding_controller.go:283]: we should make sure Get operation is successful.

**Pull request merged**